share/config.py

```diff
@@ -264,3 +272,107 @@ def parse_config(config_yaml: str) -> Config:
         conf.add_input(current_input)
 
     return conf
+
+
+def aws_sm_expander(config_yaml: str) -> str:
+    needed_secret_arns = {}
+    needed_secret_keys = {}
+
+    re_pattern = r"arn:aws:secretsmanager:(?:[^:]*):(?:[^:]*):secret:(?:[^\n]*)"
+    found_secrets = re.findall(re_pattern, config_yaml)
+
+    for secret in found_secrets:
+        secret = secret.strip('"').strip("'")
+        splitted_secret = secret.split(":")
+
+        if len(splitted_secret) != 7 and len(splitted_secret) != 8:
+            raise SyntaxError("Invalid arn format: {}".format(secret))
+
+        if secret not in needed_secret_arns:
+            needed_secret_arns[secret] = ""
+
+        region = splitted_secret[3]
+        secrets_manager_name = splitted_secret[6]
+
+        if region == "":
+            raise ValueError("Must be provided region in arn: {}".format(secret))
+
+        if secrets_manager_name == "":
+            raise ValueError("Must be provided secrets manager name in arn: {}".format(secret))
+
+        if region not in needed_secret_keys:
+            needed_secret_keys[region] = {}
+
+        if secrets_manager_name not in needed_secret_keys[region]:
+            needed_secret_keys[region][secrets_manager_name] = ""
+
+    for region in needed_secret_keys:
+        for secrets_manager_name in needed_secret_keys[region]:
+            str_secrets = get_secret_values(secrets_manager_name, region)
+            parsed_secrets = parse_secrets_str(str_secrets)
+
+            needed_secret_keys[region][secrets_manager_name] = parsed_secrets
+
+    for secret in needed_secret_arns:
+        splitted_secret_string = secret.split(":")
+
+        region = splitted_secret_string[3]
+        secrets_manager_name = splitted_secret_string[6]
+
+        if len(splitted_secret_string) == 8:
+            wanted_key = splitted_secret_string[-1]
+
+            for got_key, got_value in needed_secret_keys[region][secrets_manager_name].items():
+
+                if got_key == wanted_key:
+                    needed_secret_arns[secret] = got_value
+
+        else:
+            needed_secret_arns[secret] = needed_secret_keys[region][secrets_manager_name]
+
+    for secret in needed_secret_arns:
+        if needed_secret_arns[secret] == "":
+            raise ValueError("Value for key: {} must not be empty".format(secret))
+
+        config_yaml = config_yaml.replace(secret, needed_secret_arns[secret])
+
+    return config_yaml
+
+
+def get_secret_values(secret_name: str, region_name: str) -> str:
+    session = boto3.session.Session()
+    client = session.client(service_name="secretsmanager", region_name=region_name)
```

Comment:
this probably has to be moved to a function in order to be able to mock in tests

Reply:
should i add a separate function `get_aws_client(service_name: str, region_name: str) -> botocore.client.BaseClient` to handlers.aws.utils.py ? or do you want it somewhere above the get_secret_values?

Reply:
the implementation details are up to you

consider that in the unit tests and integration tests you will need to mock the client (see https://github.com/elastic/elastic-serverless-forwarder/blob/c85627cf4dbeb593ffd95868c647164634106487/tests/handlers/aws/test_handler.py#L260)

just for consistency instead of `session.Client` I would use https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager.html

Reply:
I will look into it!